be/src/vec/olap/olap_data_convertor.cpp

```diff
@@ -74,6 +73,39 @@ void OlapBlockDataConvertor::add_column_data_convertor(const TabletColumn& colum
     _convertors.emplace_back(create_olap_column_data_convertor(column));
 }
 
+OlapBlockDataConvertor::OlapColumnDataConvertorBaseUPtr
+OlapBlockDataConvertor::create_map_convertor(const TabletColumn& column) {
+    const auto& key_column = column.get_sub_column(0);
+    const auto& value_column = column.get_sub_column(1);
+    return std::make_unique<OlapColumnDataConvertorMap>(key_column, value_column);
+}
+
+OlapBlockDataConvertor::OlapColumnDataConvertorBaseUPtr
+OlapBlockDataConvertor::create_array_convertor(const TabletColumn& column) {
+    const auto& sub_column = column.get_sub_column(0);
+    return std::make_unique<OlapColumnDataConvertorArray>(
+            create_olap_column_data_convertor(sub_column));
+}
+
+OlapBlockDataConvertor::OlapColumnDataConvertorBaseUPtr
+OlapBlockDataConvertor::create_agg_state_convertor(const TabletColumn& column) {
+    auto data_type = DataTypeFactory::instance().create_data_type(column);
+    const auto* agg_state_type = assert_cast<const vectorized::DataTypeAggState*>(data_type.get());
+    auto type = agg_state_type->get_serialized_type()->get_type_as_type_descriptor().type;
+
+    if (type == PrimitiveType::TYPE_STRING) {
+        return std::make_unique<OlapColumnDataConvertorVarChar>(false);
+    } else if (type == PrimitiveType::TYPE_OBJECT) {
+        return std::make_unique<OlapColumnDataConvertorBitMap>();
+    } else if (type == PrimitiveType::INVALID_TYPE) {
```

Comment:
"""
The logic appears rather strange and requires a significant amount of commentary.
"""